npep/npep-127.md

```diff
@@ -0,0 +1,91 @@
+
+## Goals
+
+* Implement ingress traffic control from external destinations (outside the cluster)
+
+## Non-Goals
+
+(What is out of scope for this proposal.)
+
+## Introduction
+
+Currently, (B)ANP only allows to specify [peers](https://github.com/kubernetes-sigs/network-policy-api/blob/46be90096f23934f56342bc408b94a1aa79a159c/apis/v1alpha1/shared_types.go#L98-L103) 
+based on pod and namespace selector, that will only apply to non-hostNetwork cluster pods. This enhancement aims to
+add more types of incoming connections that may be specified. A similar effort exists for the [egress](https://github.com/kubernetes-sigs/network-policy-api/issues/126)
+direction.
+
+By default, cluster workloads are not accessible from outside the cluster, but there are many ways to expose these workloads
+(either with k8s APIs like [Ingress API](https://kubernetes.io/docs/concepts/services-networking/ingress/) or
+[Gateway API](https://gateway-api.sigs.k8s.io/), or with downstream implementations,
+like [Openshift's Route CR](https://docs.openshift.com/container-platform/4.13/rest_api/network_apis/route-route-openshift-io-v1.html)).
+Some of the APIs may provide functionality to filter incoming connections, but Network Policy API is different, because
+it is applied to the cluster workloads, and should provide the required level of security regardless of how/if a cluster workload
+is exposed to the outer world.
+
+### Use cases
+Most of the use cases refer to ANP priority, meaning the policy should be non-overridable by namespace owners, unless
+other clarification are mentioned in the use case itself.
+
+ - zero trust policy: As a cluster administrator I want to set up default deny all ingress policy as a BANP.
+
+    Example: when implementing a zero trust policy I want to make sure no ingress connection (unless explicitly allowed) will
+get to the cluster workloads. This should apply not only to the connections coming from the other cluster workloads, but to
+all incoming connections.
+
+ - block well-known ports: As a cluster administrator I want to block all ingress traffic on specific ports.
+
+   Example: when defining a cluster security policy, we decided that some well-known services like ftp, telnet, SNMP, etc.
+should not be allowed for cluster workloads. To implement this policy I want to explicitly deny all ingress connections
+for TCP ports 21, 23, 161, etc.
+
+ - compromised node protection: As a cluster administrator I want to make sure a compromised node can't send traffic to the 
+cluster workloads scheduled on the other nodes.
+
+    Example: my cluster has a set of nodes with very sensitive workloads, I want to make sure that if a worker node A is
+compromised, it will not be able to affect other worker nodes. To do so, I want to create a cluster-wide policy to
+deny ingress connections from some of the cluster nodes. I may also need to explicitly allow access from the control plane nodes.
```

Comment:
Hm... so are there pods on the compromised node? Can't the attacker forge connections from pod IPs rather than the node IP? Or is it assumed that that will be blocked as well?

Reply:
this user story considers that all pods from the same namespace are on the same node, and that inter-namespace communication is also denied. I omitted it since we already have the means to implement this part with namespace selectors (and potentially tenancy is you label namespace with nodes), but maybe it makes sense to add these details